cardano-testnet/test/cardano-testnet-golden/Test/Golden/Testnet/Config.hs

```diff
@@ -17,38 +21,40 @@ import           Testnet.Options
 import           Hedgehog
 import           Hedgehog.Extras.Test.Base (propertyOnce)
 import           Hedgehog.Extras.Test.Golden (diffVsGoldenFile)
-
+import qualified Hedgehog.Extras.Test.Process as H
 
 goldenDefaultConfigYaml :: Property
 goldenDefaultConfigYaml = propertyOnce $ do
-  let allEras = map createConfigStringAndPath [minBound..maxBound]
+  base <- H.getProjectBase
+  let allEras = map (createConfigStringAndPath base) [minBound..maxBound]
   mapM_ (uncurry diffVsGoldenFile) allEras
- where
-  createConfigStringAndPath :: AnyCardanoEra -> (String, FilePath)
-  createConfigStringAndPath era =
-    let configStr = createConfigYamlString era
-        configPath = createGoldenFilePath era
-    in (configStr, configPath)
-
-  createGoldenFilePath :: AnyCardanoEra -> FilePath
-  createGoldenFilePath (AnyCardanoEra ByronEra) =
-    "test/cardano-testnet-test/files/golden/byron_node_default_config.json"
-  createGoldenFilePath (AnyCardanoEra ShelleyEra) =
-    "test/cardano-testnet-test/files/golden/shelley_node_default_config.json"
-  createGoldenFilePath (AnyCardanoEra AllegraEra) =
-    "test/cardano-testnet-test/files/golden/allegra_node_default_config.json"
-  createGoldenFilePath (AnyCardanoEra MaryEra) =
-    "test/cardano-testnet-test/files/golden/mary_node_default_config.json"
-  createGoldenFilePath (AnyCardanoEra AlonzoEra) =
-    "test/cardano-testnet-test/files/golden/alonzo_node_default_config.json"
-  createGoldenFilePath (AnyCardanoEra BabbageEra) =
-    "test/cardano-testnet-test/files/golden/babbage_node_default_config.json"
-  createGoldenFilePath (AnyCardanoEra ConwayEra) =
-    "test/cardano-testnet-test/files/golden/conway_node_default_config.json"
-
-  createConfigYamlString :: AnyCardanoEra -> String
-  createConfigYamlString era =
-    let configBs = LB.toStrict . encodePretty
-                   . Object . defaultYamlHardforkViaConfig $ era
-        configStr = Text.unpack $ Text.decodeUtf8 configBs
-    in configStr
+
+createConfigStringAndPath :: FilePath -> AnyCardanoEra -> (String, FilePath)
+createConfigStringAndPath base era =
+  let configStr = createConfigYamlString era
+      configPath = createGoldenFilePath base era
+  in (configStr, configPath)
+
+createGoldenFilePath :: FilePath -> AnyCardanoEra -> FilePath
+createGoldenFilePath base = \case
```

Comment:
You can remove the directory path from case expression and just return the file name from it and glue the path outside of it. It may be more readable.

Reply:
done